lib/main.rb

```diff
@@ -0,0 +1,82 @@
+module Enumerable
+  def my_each
+    element = 0
+    while element < length
+      yield(self[element])
+      element += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    element = 0
+    while element < length
+      yield(self[element], element)
+      element += 1
+    end
+  end
+
+  def my_select
+    new_array = []
+    my_each do |a|
+      yield(a) ? new_array.push(a) : new_array
+    end
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] my_select returns an array containing all elements of enum for which the given block returns a true value
  ```ruby
  block = proc {|x| x>4}
  array = [1,2,5,8,7,98,4]
  p array.my_select(&block) == array.select(&block) # false
- [ ] my_select returns an enumerator if no block is given
  ```ruby
  array = [1,2,5,8,7,98,4]
  p array.my_select.class # not Enumerator